File: src/main/java/nl/mpcjanssen/simpletask/sort/CreationDateComparator.java
package nl.mpcjanssen.simpletask.sort;

import com.google.common.collect.Ordering;
import hirondelle.date4j.DateTime;
import nl.mpcjanssen.simpletask.task.Task;
import nl.mpcjanssen.simpletask.util.Strings;

public class CreationDateComparator extends Ordering<Task> {

    @Override
    public int compare(Task a, Task b) {
        int result = 0;
        if (Strings.isEmptyOrNull(a.getCreateDate()) && Strings.isEmptyOrNull(b.getCreateDate())) {
            result = 0;
        } else if (Strings.isEmptyOrNull(a.getCreateDate())) {
            result = 1;
        } else if (Strings.isEmptyOrNull(b.getCreateDate())) {
            result = -1;
        } else {
            DateTime dateA = new DateTime(a.getCreateDate());
            DateTime dateB = new DateTime(b.getCreateDate());
            result = dateA.compareTo(dateB);
        }
        return result;
    }
}

File: src/main/java/nl/mpcjanssen/simpletask/sort/CreationDateComparator.kt
package nl.mpcjanssen.simpletask.sort

import nl.mpcjanssen.simpletask.task.Task
import java.util.*

class CreationDateComparator : Comparator<Task> {

    override fun compare(a: Task?, b: Task?): Int {
        if (a === b) {
            return 0
        } else if (a == null) {
            return -1
        } else if (b == null) {
            return 1
        }
        val aDate = a.createDate
        val bDate =  b.createDate
        if ( aDate == null && bDate == null) {
            return 0
        } else if (aDate == null) {
            return 1
        } else if (bDate == null) {
            return  -1
        }
        return aDate.compareTo(bDate)
    }
}
